### PR TITLE
Add description to all built-in function args and return values

### LINKED ISSUE #7151: Use `.Description` to document all built-in function args and return types consistently
Most of the OPA built-in functions include some description for each argument, and what the function returns.

Example:

```go
var CryptoHmacEqual = &Builtin{
	Name:        "crypto.hmac.equal",
	Description: "Returns a boolean representing the result of comparing two MACs for equality without leaking timing information.",
	Decl: types.NewFunction(
		types.Args(
			types.Named("mac1", types.S).Description("mac1 to compare"),
			types.Named("mac2", types.S).Description("mac2 to compare"),
		),
		types.Named("result", types.B).Description("`true` if the MACs are equals, `false` otherwise"),
	),
}
```

However, a number of built-in functions don't have any description for these attributes:

```go
var ProvidersAWSSignReqObj = &Builtin{
	Name:        "providers.aws.sign_req",
	Description: "Signs an HTTP request object for Amazon Web Services. Currently implements [AWS Signature Version 4 request signing](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html) by the `Authorization` header method.",
	Decl: types.NewFunction(
		types.Args(
			types.Named("request", types.NewObject(nil, types.NewDynamicProperty(types.S, types.A))),
			types.Named("aws_config", types.NewObject(nil, types.NewDynamicProperty(types.S, types.A))),
			types.Named("time_ns", types.N),
		),
		types.Named("signed_request", types.NewObject(nil, types.NewDynamicProperty(types.A, types.A))),
	),
	Categories: providersAWSCat,
}
```

Since these values are used both in the OPA docs and by tools like Regal for in-editor tooltips, it would be great if we took the time to fix so that all built-in functions are consistently documenting their args and return values.

All the built-in functions are conveniently in https://github.com/open-policy-agent/opa/blob/main/ast/builtins.go

### PR DESCRIPTION
**Pull request by anderseknert**
Fixes #7151

### CODE REVIEW

File: ast/builtins.go [hunk 1]
```diff
@@ -758,13 +758,13 @@ var Sum = &Builtin{
 
 var Product = &Builtin{
 	Name:        "product",
-	Description: "Muliplies elements of an array or set of numbers",
+	Description: "Multiplies elements of an array or set of numbers",
```
Comment (srenatus) [inline ast/builtins.go:761]: 😆 Yay

File: ast/builtins.go [hunk 2]
```diff
@@ -758,13 +758,13 @@ var Sum = &Builtin{
...
 	Decl: types.NewFunction(
 		types.Args(
 			types.Named("collection", types.NewAny(
 				types.NewSet(types.N),
 				types.NewArray(nil, types.N),
-			)),
+			)).Description("the set or array of "),
```
Comment (srenatus) [inline ast/builtins.go:767]: ...what?

**Review by srenatus — APPROVED** [review]

File: ast/builtins.go (hunk 2 above)
Comment (anderseknert) [reply ast/builtins.go:767]: What a suspense! I'll fix, thanks!

Comment (anderseknert) [reply ast/builtins.go]: done

**Review by srenatus — APPROVED** [review]

**Comment by srenatus** [comment]
Hmm why does it not have an effect [here](https://deploy-preview-7153--openpolicyagent.netlify.app/docs/edge/policy-reference/#builtin-sets-or)?
image

**Pull request merged**